.github/workflows/commit-validation.yml

```diff
@@ -6,15 +6,49 @@
 name: "Commit message validation"
 on:
   pull_request:
-    branches: [main, commitlint]
-    paths: [
-      # Commit validation in PRs applies to build-tools only as of now.
-      'build-tools/**', '.github/workflows/commit-validation.yml']
+    branches: [main, next]
     types: [opened, synchronize, edited, ready_for_review]
 jobs:
+  # This job determines what paths have changed in the PR and outputs a boolean
```

Comment:
It would probably be worth adding a brief description of why we can't just use the built-in `paths` parameter and have to use this instead.